server/app/views/admin/programs/ProgramBlockReadOnlyView.java

```diff
@@ -0,0 +1,312 @@
+        blockDefinition.programQuestionDefinitions();
+    DivTag programQuestions = div();
+    IntStream.range(0, blockQuestions.size())
+        .forEach(
+            questionIndex -> {
+              programQuestions.with(renderQuestion(program, blockDefinition, questionIndex));
+            });
+    return new ArrayList<DomContent>(
+        Arrays.asList(blockInfoDisplay, predicateDisplay, programQuestions));
+  }
+
+  protected DivTag renderPredicate(
+      ProgramDefinition programDefinition /* needed for subclasses */,
+      BlockDefinition blockDefinition,
+      ImmutableList<QuestionDefinition> questions) {
+
+    Optional<PredicateDefinition> predicate = blockDefinition.visibilityPredicate();
+    String currentBlockStatus =
+        predicate.isEmpty()
+            ? "This screen is always shown."
+            : predicate.get().toDisplayString(blockDefinition.name(), questions);
+
+    return div()
+        .withClasses("my-4")
+        .with(div("Visibility condition").withClasses("text-lg", "font-bold", "py-2"))
+        .with(div(currentBlockStatus).withClasses("text-lg", "max-w-prose"));
+  }
+
+  /** Renders an individual question to be shown in the Block panel. */
+  protected DivTag renderQuestion(
+      ProgramDefinition programDefinition /* needed for subclasses */,
+      BlockDefinition blockDefinition,
+      int questionIndex) {
+    ImmutableList<ProgramQuestionDefinition> blockQuestions =
+        blockDefinition.programQuestionDefinitions();
+    ProgramQuestionDefinition question = blockQuestions.get(questionIndex);
+
+    QuestionDefinition questionDefinition = question.getQuestionDefinition();
+    DivTag ret =
+        div()
+            .withClasses(
+                ReferenceClasses.PROGRAM_QUESTION,
+                "my-2",
+                "border",
+                "border-gray-200",
+                "px-4",
+                "py-2",
+                "flex",
+                "gap-4",
+                "items-center",
+                StyleUtils.hover("text-gray-800", "bg-gray-100"));
+
+    SvgTag icon =
+        Icons.questionTypeSvg(questionDefinition.getQuestionType())
+            .withClasses("shrink-0", "h-12", "w-6");
+    String questionHelpText =
+        questionDefinition.getQuestionHelpText().isEmpty()
+            ? ""
+            : questionDefinition.getQuestionHelpText().getDefault();
+    DivTag content =
+        div()
+            .withClass("flex-grow")
+            .with(
+                p(questionDefinition.getQuestionText().getDefault()),
+                p(questionHelpText).withClasses("mt-1", "text-sm"),
+                p(String.format("Admin ID: %s", questionDefinition.getName()))
+                    .withClasses("mt-1", "text-sm"));
+    return ret.with(icon, content);
+  }
+
+  @Override
+  protected String getEditButtonText() {
+    return "Edit program";
+  }
+
+  // TODO when the ProgramBlockReadOnlyView is used, clicking the button should lead to the
```

Comment:
There is a range of other clicking behaviours that also need to be implemented and I'm planning to group all of those together as it groups together similar tasks and will make testing easier.

The three changes for the higher level task are:
Make a flag (you reviewed this)
Refactor so that a view only version is possible (this change, not user visible)
Implement and link the UI elements that are missing and protect with flag

I've linked the TODO to the higer level bug that tracks the change TODO(#3162).